pkg/ebpf/tracee.go

```diff
@@ -408,17 +413,62 @@ func (t *Tracee) Init() error {
 		return fmt.Errorf("error creating process tree: %v", err)
 	}
 
-	// Initialize containers enrichment logic
+	// Initialize cgroup filesystems
 
-	capabilities.GetInstance().Requested(func() error { // TODO: workaround until PR: #2233 is in place
+	var defaultCgroup cgroup.Cgroup
 
-		t.containers, err = containers.New(t.config.Sockets, "containers_map", t.config.Debug)
+	err = caps.Requested(func() error { // ring2
```

Comment:
So I think all the nested logic here can go into a function in the new `cgroup` package of `cgroup.New()` that will return an instance of the `Cgroup` interface, that way we may be able to not have two instances of cgroup at the same time on tracee's struct level.
It might be necessary to add a "combined" v1+v2 cgroup struct impl for this, in which case it may be better to just leave it be.